editoast/src/views/timetable/train_schedule.rs

```diff
@@ -590,6 +609,309 @@ async fn occupancy_blocks(
+            None => Duration::milliseconds(0).try_into().unwrap(),
+        };
+
+        // Get the index of the path item in the list of path items that corresponds to the operational point
+        let index = train_schedule
+            .path
+            .iter()
+            .position(|path| path.id == path_item_id)
+            .expect("Unexcpected path item");
+        let position = path_item_positions[index];
+        let time = report_train.path_item_times[index];
+
+        // Retrieve the track section that matches the path item's position along the path
+        let track_section = match path_projection.get_location(position) {
+            TrackLocationFromPath::One(track_offset) => operational_point_track_offsets
+                .iter()
+                .find(|to| *to == &track_offset)
+                .expect("Invalid track offset")
+                .track
+                .clone(),
+            TrackLocationFromPath::Two(track_offset, track_offset1) => {
+                operational_point_track_offsets
+                    .iter()
+                    .find(|to| *to == &track_offset)
+                    .or(operational_point_track_offsets
+                        .iter()
+                        .find(|to| *to == &track_offset1))
+                    .expect("Invalid track offset")
+                    .track
+                    .clone()
+            }
+        };
+
+        Some((
+            track_section.to_string(),
+            TrackOccupancy {
+                time_begin: train_schedule.start_time + Duration::milliseconds(time as i64),
+                duration: stop_duration,
+                train_schedule_id: train_schedule.id,
+            },
+        ))
+    } else {
+        let mut track_and_time = None;
+
+        // Iterate over the track offsets linked to the operational point
+        // until we find its corresponding position on the pathfinding route.
+        // Then, retrieve the associated time (via interpolation) and position on the path.
+        for track_offset in operational_point_track_offsets.iter() {
+            if let Some(position) = path_projection.get_position(track_offset) {
+                let time = interpolated_time_at_position(
+                    &report_train.positions,
+                    &report_train.times,
+                    position,
+                );
+                track_and_time = Some((track_offset.track.clone(), time));
+                break;
+            }
+        }
+        if let Some((track_section, time)) = track_and_time {
+            Some((
+                track_section.to_string(),
+                TrackOccupancy {
+                    time_begin: train_schedule.start_time + Duration::milliseconds(time as i64),
+                    duration: Duration::milliseconds(0).try_into().unwrap(),
+                    train_schedule_id: train_schedule.id,
+                },
+            ))
+        } else {
+            None
+        }
+    }
```

Comment:
I think we can simplify this part:

```suggestion
    } else {
        // Iterate over the track offsets linked to the operational point
        // until we find its corresponding position on the pathfinding route.
        // Then, retrieve the associated time (via interpolation) and position on the path.
        for track_offset in operational_point_track_offsets.iter() {
            if let Some(position) = path_projection.get_position(track_offset) {
                let time = interpolated_time_at_position(
                    &report_train.positions,
                    &report_train.times,
                    position,
                );

                return Some((
                    track_offset.track.clone(),
                    TrackOccupancy {
                        time_begin: train_schedule.start_time + Duration::milliseconds(time as i64),
                        duration: Duration::milliseconds(0).try_into().unwrap(),
                        train_schedule_id: train_schedule.id,
                    },
                ));
            }
        }
       return None;
    }
```